frontend/server/src/main/java/org/pytorch/serve/wlm/SequenceBatchAggregator.java

```diff
@@ -0,0 +1,213 @@
+package org.pytorch.serve.wlm;
+
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.pytorch.serve.job.Job;
+import org.pytorch.serve.job.JobGroup;
+import org.pytorch.serve.util.messages.BaseModelRequest;
+import org.pytorch.serve.util.messages.ModelWorkerResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SequenceBatchAggregator extends BatchAggregator {
+
+    private static final Logger logger = LoggerFactory.getLogger(SequenceBatchAggregator.class);
+    private ExecutorService pollExecutors;
+    private LinkedBlockingDeque<String> eventJobGroupIds;
+    private LinkedBlockingDeque<Job> jobQueue;
+    private Thread eventDispatcher;
+    private AtomicBoolean isPollJobGroup;
+    private LinkedList<String> currentJobGroupIds;
+
+    public SequenceBatchAggregator(Model model) {
+        super(model);
+        this.currentJobGroupIds = new LinkedList<>();
```

Comment:
Would be good to write a quick comment on the purpose of each queue and the flow if the ids between them.